explorer/client/src/components/search/Search.tsx

```diff
@@ -1,85 +1,183 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import React, { useState, useCallback, useContext } from 'react';
+import React, {
+    useState,
+    useCallback,
+    useContext,
+    useRef,
+    useEffect,
+} from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import { ReactComponent as SearchIcon } from '../../assets/search.svg';
 import { NetworkContext } from '../../context';
-import { navigateWithUnknown } from '../../utils/searchUtil';
+import {
+    navigateWithCategory,
+    SEARCH_CATEGORIES,
+} from '../../utils/searchUtil';
+import { type ResultType } from './SearchResultType';
+import SearchResults from './SearchResults';
 
 import styles from './Search.module.css';
 
 function Search() {
-    const [input, setInput] = useState('');
-    const [network] = useContext(NetworkContext);
     const navigate = useNavigate();
+    const wrapperref = useRef<HTMLDivElement>(null);
+    const [network] = useContext(NetworkContext);
+    const [input, setInput] = useState('');
+
+    const [result, setResult] = useState<ResultType[] | null>(null);
+    const [resultIndex, setResultIndex] = useState(0);
+
+    const handleClickOutside = (event: MouseEvent): void => {
+        if (
+            wrapperref.current &&
+            !wrapperref.current.contains(event.target as Node)
+        ) {
+            setResult(null);
+            setInput('');
+        }
+    };
+
+    const handleKeyPress = useCallback(
+        (event: KeyboardEvent): void => {
+            // Press Down Key or Tab
+            if (event.keyCode === 40 || event.keyCode === 9) {
```

Comment:
If we do end up keeping this logic, use `event.key` which is the non-deprecated property.

Reply:
Good idea - done. The implementation follows guidance from Mozilla (https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key#examples) to cover most web browsers.